botnim/sync.py

```diff
@@ -65,9 +66,35 @@ def update_assistant(config, config_dir, production, backend, replace_context=Fa
         ## Elasticsearch
         elif backend == 'es':
             vs = VectorStoreES(config, config_dir, es_host, es_username, es_password, production=production)
-        # Update the vector store with the context
-        tools, tool_resources = vs.vector_store_update(config['context'], replace_context)
-    
+            # Add explicit Elasticsearch vector search tools for each context
+            tools = []
+            
+            # Only process context if replace_context is True or if it's the first time
+            if replace_context:
+                for context in config['context']:
+                    context_name = context['name']
+                    bot_name = config_dir.name
+                    elastic_tool = create_elastic_search_tool(
```

Comment:
the entire point of creating the 'VectorStore' class was to encapsulate the logic for each of the vector store backends and allow sync.py to work with both of them.
Now you've added elasticsearch specific code to sync.py, which defeats that purpose.

Also - the tools array should be populated always, not only when replace_context = True.